django/db/models/base.py

```diff
@@ -674,7 +674,10 @@ def __setstate__(self, state):
 
     def _get_pk_val(self, meta=None):
         meta = meta or self._meta
-        return getattr(self, meta.pk.attname)
+        try:
+            return getattr(self, meta.pk.attname)
+        except AttributeError:
+            return None
```

Comment:
This may need to be removed and dealt with further up the stack... for databases that don't support `RETURNING` like mysql then a generated pk shouldn't return `None` but instead continue to raise `AttributeError`.